artifacts/reader.py

```diff
@@ -4,17 +4,58 @@
 import abc
 import glob
 import os
+import yaml
 
 from artifacts import artifact
 from artifacts import definitions
 from artifacts import errors
 
-import yaml
-
 
-class ArtifactsReader(object):
+class BaseArtifactsReader(object):
   """Class that implements the artifacts reader interface."""
 
+  LABELS = None
+  SUPPORTED_OS = None
+
+  @property
+  def supported_os(self):
```

Comment:
why define these as class constants?
why would the read need to define these properties?

let me if I understand what you want to achieve. Is your goal to allow to e.g. add custom labels?